src/lib.rs

```diff
@@ -106,16 +107,166 @@ impl Build {
     }
 }
 
+#[derive(Debug, Clone, StructOpt)]
+pub struct Watch {
+    #[structopt(long = "watch", short = "w")]
+    watch_paths: Vec<PathBuf>,
+    #[structopt(long = "ignore", short = "i")]
+    exclude_paths: Vec<PathBuf>,
+    #[structopt(skip)]
+    workspace_exclude_paths: Vec<PathBuf>,
+}
+
+impl Watch {
+    pub fn new() -> Self {
+        Self {
+            exclude_paths: Vec::new(),
+            watch_paths: Vec::new(),
+            workspace_exclude_paths: Vec::new(),
+        }
+    }
+
+    pub fn exclude(&mut self, path: impl AsRef<Path>) {
+        self.exclude_paths.push(path.as_ref().to_path_buf())
+    }
+
+    pub fn excludes(&mut self, paths: Vec<impl AsRef<Path>>) {
```

Comment:
The Vec here should be an impl Iterator or IntoIterator. Check the trait Extend to see how to use that in argument